### PR TITLE
Fix/payment layer critical bugs

### PR DESCRIPTION
**Pull request by parshvi1508**
## What does this PR do?

Completes the P0 Phase 3 Security and Multi-Tenant Stabilization fixes.
- Introduces `SUPERADMIN` role and prevents privilege escalation on `/register`.
- Scopes all `AdminServices` (Users, Bookings, Analytics) queries by `tenant_id` to prevent cross-tenant data leaks.
- Rewrites dashboard analytics to query real Postgres data instead of mock variables.
- Fixes Redis connection leak in `core/security.py` by sharing the core pool.
- Adds missing idempotent processing logic for Razorpay webhooks (`process_webhook`).
- Pins `requirements.txt` to strictly locked versions to prevent Azure App Service build failures.

## Type of change
- [x] Security Fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?
- [x] Run `alembic stamp 005` followed by `alembic upgrade head`
- [x] Run `python scripts/promote_superadmin.py <your-email>` to grant platform ownership.
- [x] Run `pytest` to execute the edge-case test suite (`test_auth_roles`, `test_tenant_scoping`, `test_webhook_idempotency`).

## Checklist:
- [x] My code follows the strict DDD architecture and rules in `API_REFERENCE.md`
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (`README.md`)
- [x] My changes generate no new warnings (MyPy passes)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/auth/services.py [hunk 1]
```diff
@@ -296,6 +317,9 @@ async def verify_otp_and_reset_password(self,email:str,otp:str,new_password:str)
             )
         
         if stored_otp != otp:
+            # On 5th failed attempt, delete OTP to force re-request
+            if attempts + 1 >= 5:
+                await redis.delete(key)
```
[the remark this answers is not in the document]
Comment (parshvi1508) [reply app/auth/services.py]: @copilot apply changes based on [this feedback](https://github.com/Truism-com/Trurism_Backend/pull/99#discussion_r3248289470)

File: app/wallet/services.py [hunk 2]
```diff
@@ -624,10 +638,10 @@ async def release_hold(
...
         if self.redis:
             if not convert_to_debit:
-                await self.redis.delete(f"wallet:hold:{hold_id}")
+                await self.redis.delete(f"hold:{hold_id}")
             else:
                 hold["status"] = "converted"
-                await self.redis.set(f"wallet:hold:{hold_id}", json.dumps(hold))
+                await self.redis.set(f"hold:{hold_id}", json.dumps(hold))
         
         await self.db.commit()
```
[the remark this answers is not in the document]
Comment (parshvi1508) [reply app/wallet/services.py:646]: @copilot apply changes based on [this feedback](https://github.com/Truism-com/Trurism_Backend/pull/99#discussion_r3248289429)

File: app/booking/services.py [hunk 3]
```diff
@@ -395,6 +395,43 @@ async def get_user_hotel_bookings(self, user_id: int, skip: int = 0, limit: int
...
+                amount=refund_amount,
+                booking_id=booking_id,
+                booking_type="hotel",
+                original_payment_method=booking.payment_method,
+                reason=cancel_request.reason
+            )
+        
+        booking.status = BookingStatus.CANCELLED
+        booking.refund_amount = refund_amount
+        booking.payment_status = PaymentStatus.REFUNDED if refund_amount > 0 else PaymentStatus.FAILED
```
[the remark this answers is not in the document]
Comment (parshvi1508) [reply app/booking/services.py]: @copilot apply changes based on [this feedback](https://github.com/Truism-com/Trurism_Backend/pull/99#discussion_r3248289271)

File: app/wallet/models.py [hunk 4]
```diff
@@ -70,24 +71,24 @@ class Wallet(Base, TenantMixin):
...
-    daily_transaction_limit: Mapped[float] = mapped_column(default=100000.0)  # Max daily transactions
-    min_balance: Mapped[float] = mapped_column(default=0.0)  # Minimum balance to maintain
+    daily_transaction_limit: Mapped[Decimal] = mapped_column(Numeric(14, 2), default=100000.0)  # Max daily transactions
+    min_balance: Mapped[Decimal] = mapped_column(Numeric(14, 2), default=0.0)  # Minimum balance to maintain
     
     # Statistics
-    total_credited: Mapped[float] = mapped_column(default=0.0)
-    total_debited: Mapped[float] = mapped_column(default=0.0)
+    total_credited: Mapped[Decimal] = mapped_column(Numeric(14, 2), default=0.0)
+    total_debited: Mapped[Decimal] = mapped_column(Numeric(14, 2), default=0.0)
```
[the remark this answers is not in the document]
Comment (parshvi1508) [reply app/wallet/models.py]: @copilot apply changes based on [this feedback](https://github.com/Truism-com/Trurism_Backend/pull/99#discussion_r3248289349)

File: migrations/versions/012_composite_unique_booking_reference.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            ['booking_reference', 'tenant_id']
+        )
+
+    # BusBooking: drop old unique, add composite
+    with op.batch_alter_table('bus_bookings') as batch_op:
+        batch_op.drop_constraint('bus_bookings_booking_reference_key', type_='unique')
+        batch_op.create_unique_constraint(
+            'uq_bus_booking_ref_tenant',
+            ['booking_reference', 'tenant_id']
+        )
```
[the remark this answers is not in the document]
Comment (parshvi1508) [reply migrations/versions/012_composite_unique_booking_reference.py:58]: @copilot apply changes based on [this feedback](https://github.com/Truism-com/Trurism_Backend/pull/99#discussion_r3248289375)

**Comment by parshvi1508** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/Truism-com/Trurism_Backend/pull/99#pullrequestreview-4298233062)

**Comment by parshvi1508** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/Truism-com/Trurism_Backend/pull/99#pullrequestreview-4298233062)

**Comment by parshvi1508** [comment]
@copilot apply changes based on the comments in [this thread](https://github.com/Truism-com/Trurism_Backend/pull/99#pullrequestreview-4306340299)

**Review by parshvi1508 — COMMENTED** [review]
done
